tests/drone/skipIfUnchanged.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+CHANGED_UNIT_TESTS_ONLY=True
+CHANGED_DOCS_ONLY=True
+CHANGED_UNIT_TESTS_AND_DOCS_ONLY=True
+
+DIFFS=$(git diff origin/master --name-only)
+DOCS_ONLY_CHANGES=$(echo "$DIFFS" | grep -E '(^docs/.*|^changelog/.*|^.drone.star|^tests/drone/skipIfUnchanged.sh)')
+UNIT_TESTS_ONLY_CHANGES=$(echo "$DIFFS" | grep '^packages/.*/tests/.*')
+DOCS_AND_UNIT_TESTS_ONLY_CHANGES=$(echo "$DIFFS" | grep -E '(^docs/.*|^changelog/.*|^packages/.*/tests/.*|^.drone.star|^tests/drone/skipIfUnchanged.sh)')
+
+changesExceptDocsChanges=$(echo ${DIFFS[@]} ${DOCS_ONLY_CHANGES[@]} | tr ' ' '\n' | sort | uniq -u)
+changesExceptUnitTestsChanges=$(echo ${DIFFS[@]} ${UNIT_TESTS_ONLY_CHANGES[@]} | tr ' ' '\n' | sort | uniq -u)
+changesExceptUnitTestsAndDocsChanges=$(echo ${DIFFS[@]} ${DOCS_AND_UNIT_TESTS_ONLY_CHANGES[@]} | tr ' ' '\n' | sort | uniq -u)
+
+if ((${#DIFFS})); then
+    if ((${#changesExceptUnitTestsAndDocsChanges})); then
+        echo "FILES OTHER THAN UNIT TESTS AND DOCS ARE ALSO CHANGED"
+        CHANGED_UNIT_TESTS_AND_DOCS_ONLY=False
+        CHANGED_UNIT_TESTS_ONLY=False
+        CHANGED_DOCS_ONLY=False
+    else
+        if ((${#changesExceptDocsChanges})); then
+            echo "FILES OTHER THAN DOCS FILES ARE ALSO CHANGED"
+            CHANGED_DOCS_ONLY=False
+            elif ((${#changesExceptUnitTestsChanges})); then
+            echo "FILES OTHER THAN UNIT TEST FILES ARE ALSO CHANGED"
+            CHANGED_UNIT_TESTS_ONLY=False
+        fi
+    fi
+
+    if [ $CHANGED_UNIT_TESTS_ONLY == "True" ]
+    then
+        echo "ONLY UNIT TEST FILES ARE CHANGED"
+        touch runUnitTestsOnly
```

Comment:
These could be refactored to just use internal variables, rather than touching a real file to pass the "answer" down to lower lines. Or even directly:
 ```suggestion
        if [ "$1" == "acceptance-tests" ]; then
            exit 78
        fi
```

Reply:
I wanted to try it quick and dirty... But my additions (`|^.drone.star|^tests/drone/skipIfUnchanged.sh`) to the git diff logic didn't work out.

I wonder if we should transform that logic into a real drone plugin.  That would help reusing it. What do you think about this? We could talk about this next week ;-)